File: sdk/js/tessrax.js
/**
 * Tessrax JS SDK v19.6 — minimal client for browser or Node.js
 */
const BASE_URL = "http://localhost:8080";

export async function emitReceipt(payload) {
  const r = await fetch(`${BASE_URL}/api/receipts`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!r.ok) throw new Error(await r.text());
  return await r.json();
}

export async function verifyReceipt(receipt) {
  const r = await fetch(`${BASE_URL}/api/verify`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(receipt),
  });
  if (!r.ok) throw new Error(await r.text());
  const { verified } = await r.json();
  return verified;
}

export async function getStatus(window = 100) {
  const r = await fetch(`${BASE_URL}/api/status?window=${window}`);
  if (!r.ok) throw new Error(await r.text());
  return await r.json();
}

File: sdk/go/tessrax/tessrax.go
// Tessrax Go SDK — v19.6
// Provides minimal client for governance API.
package tessrax

import (
	"bytes"
	"encoding/json"
	"net/http"
	"strconv"
	"time"
)

var BaseURL = "http://localhost:8080"

type Receipt map[string]interface{}

func Emit(payload Receipt) (Receipt, error) {
	buf, _ := json.Marshal(payload)
	resp, err := http.Post(BaseURL+"/api/receipts", "application/json", bytes.NewBuffer(buf))
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	var out Receipt
	json.NewDecoder(resp.Body).Decode(&out)
	return out, nil
}

func Verify(receipt Receipt) (bool, error) {
	buf, _ := json.Marshal(receipt)
	resp, err := http.Post(BaseURL+"/api/verify", "application/json", bytes.NewBuffer(buf))
	if err != nil {
		return false, err
	}
	defer resp.Body.Close()
	var out struct {
		Verified bool `json:"verified"`
	}
	json.NewDecoder(resp.Body).Decode(&out)
	return out.Verified, nil
}

func Status(window int) (Receipt, error) {
	client := http.Client{Timeout: 5 * time.Second}
	resp, err := client.Get(BaseURL + "/api/status?window=" + strconv.Itoa(window))
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	var out Receipt
	json.NewDecoder(resp.Body).Decode(&out)
	return out, nil
}
